src/python/pants/backend/terraform/target_gen_test.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import textwrap
+
+import pytest
+
+from pants.backend.terraform import target_gen
+from pants.backend.terraform.target_gen import GenerateTerraformModuleTargetsRequest
+from pants.backend.terraform.target_types import (
+    TerraformModule,
+    TerraformModules,
+    TerraformModuleSources,
+)
+from pants.build_graph.address import Address
+from pants.core.util_rules import external_tool, source_files
+from pants.engine.rules import QueryRule
+from pants.engine.target import GeneratedTargets
+from pants.testutil.rule_runner import RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    rule_runner = RuleRunner(
+        target_types=[TerraformModule, TerraformModules],
+        rules=[
+            *external_tool.rules(),
+            *source_files.rules(),
+            *target_gen.rules(),
+            QueryRule(GeneratedTargets, [GenerateTerraformModuleTargetsRequest]),
+        ],
+    )
+    rule_runner.set_options(["--backend-packages=pants.backend.experimental.terraform"])
+    return rule_runner
+
+
+def test_target_generation(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "BUILD": textwrap.dedent(
+                """\
+        terraform_modules(name="tf_mods")
+        """
+            ),
+            "src/tf/versions.tf": textwrap.dedent(
+                """\
+        terraform {
+          required_version = ">= 1.0"
+        }
+        """
```

Comment:
Nit, preferred to indent these so they line up with `"""`. It makes the test easier to read.

Reply:
Also these don't actually need content, right? Just the existence of a `.tf` file.

Reply:
I've noticed that black has a tendency to mess this indentation up, when moving around surrounding code/text, so you have to manually go in and fix it, then black will leave it alone.

Reply:
I got rid of the dedent since the content was needed.